docs/networking.md

```diff
@@ -0,0 +1,114 @@
+<!--[metadata]>
++++
+title = "Networking in Compose"
+description = "How Compose sets up networking between containers"
+keywords = ["documentation, docs,  docker, compose, orchestration, containers, networking"]
+[menu.main]
+parent="smn_workw_compose"
+weight=6
++++
+<![end-metadata]-->
+
+
+# Networking in Compose
+
+By default, Compose sets up a single default [network](http://TODO/docker-networking-docs) for your app. Each container for a service joins the default network and is discoverable via DNS under the service's name.
+
+> **Note:** Your app's network is given the same name as the "project name", which is based on the name of the directory it lives in. See the [CLI docs](cli.md#p-project-name-name) for how to override it.
+
+For example, suppose your app is in a directory called `myapp`, and your `docker-compose.yml` looks like this:
+
+    web:
+      build: .
+      ports:
+        - "8000:8000"
+    db:
+      image: postgres
+
+When you run `docker-compose up`, the following happens:
+
+1. A network called `myapp` is created.
+2. A container is created using `web`'s configuration. It joins the network `myapp` under the name `web`.
+3. A container is created using `db`'s configuration. It joins the network `myapp` under the name `db`.
+
+Each container can now look up the hostname `web` or `db` and get back the appropriate container's IP address. For example, `web`'s application code could connect to the URL `postgres://db:5432` and start using the Postgres database.
+
+Because `web` explicitly maps a port, it's also accessible from the outside world via port 8000 on your Docker host's network interface.
```

Comment:
I'll let @dave-tucker comment, but having a container join two networks was one of the scenarios we were trying to avoid to 1.8 (because it needs a decent amount of thought to get right across the broad range of plugin network drivers).  So I was expecting the `ports` directive would only do anything if the network you've specified is the `bridge` network (either because that's what is set up as the default network, or because you explicitly specified that).  For the plugin drivers (via libnetwork) the `ports` directive would be ignored and it's up to the user to additionally issue whatever commands required (if any) to make web accessible from the outside world (e.g. `weave expose` or similar in the case of Weave).